snuba/request/validation.py

```diff
@@ -105,25 +94,38 @@ def update_attribution_info(
     return attribution_info
 
 
-def build_request(
+def parse_api_request(
     body: Dict[str, Any],
-    parser: Parser,
     settings_class: Union[Type[HTTPQuerySettings], Type[SubscriptionQuerySettings]],
     schema: RequestSchema,
     dataset: Dataset,
     timer: Timer,
     referrer: str,
+    is_mql: bool,
     custom_processing: Optional[CustomProcessors] = None,
-) -> Request:
-    with sentry_sdk.start_span(description="build_request", op="validate") as span:
+) -> tuple[
+    RequestParts,
+    HTTPQuerySettings | SubscriptionQuerySettings,
+    Query | CompositeQuery[Entity],
+    str,
+]:
+    """
+    Given a snuba-api request-body (among other things), parse it into a query ast and return the ast (among other things).
```

Comment:
```
    Given a snuba-api request-body (among other things), parse it into a query ast and return the ast (among other things).
```

(among other things) is not a helpful comment and if you have to put it maybe it indicates that this interface does too many things

Reply:
We should think about separating parsing and build request in the context of querying storages directly. In the future, when an incoming "storage query" reaches snuba, would we parse the query the same way we do with an "entity query"? If not, we should be able to swap out that specific stage of the pipeline that handles parsing with something else that works for storage queries. Same goes for `build_request` (maybe this doesn't need to change). It would be nice to use the `QueryPipelineStage` interface to enable this work.